lib/node_modules/@stdlib/array/empty-like/docs/types/index.d.ts

```diff
@@ -0,0 +1,618 @@
+*
+* var x = zeros( 2, 'float32' );
+* // returns <Float32Array>[ 0.0, 0.0 ]
+*
+* var arr = emptyLike( x, 'uint32' );
+* // returns <Uint32Array>
+*/
+declare function emptyLike( x: Uint32Array ): Uint32Array;
+
+/**
+* Creates an uninitialized array having the same length as a provided input array.
+*
+* ## Notes
+*
+* -   In browser environments, the function always returns zero-filled arrays.
+* -   In Node.js versions `>=3.0.0`, the underlying memory of returned typed arrays is **not** initialized. Memory contents are unknown and may contain **sensitive** data.
+*
+* @param x - input array from which to derive the output array length
+* @param dtype - data type
+* @returns empty array
+*
+* @example
+* var zeros = require( `@stdlib/array/zeros` );
+*
+* var x = zeros( 2, 'float32' );
+* // returns <Float32Array>[ 0.0, 0.0 ]
+*
+* var arr = emptyLike( x, 'uint16' );
+* // returns <Uint16Array>
+*/
+declare function emptyLike( x: Uint16Array ): Uint16Array;
+
+/**
+* Creates an uninitialized array having the same length as a provided input array.
+*
+* ## Notes
+*
+* -   In browser environments, the function always returns zero-filled arrays.
+* -   In Node.js versions `>=3.0.0`, the underlying memory of returned typed arrays is **not** initialized. Memory contents are unknown and may contain **sensitive** data.
+*
+* @param x - input array from which to derive the output array length
+* @param dtype - data type
+* @returns empty array
+*
+* @example
+* var zeros = require( `@stdlib/array/zeros` );
+*
+* var x = zeros( 2, 'float32' );
+* // returns <Float32Array>[ 0.0, 0.0 ]
+*
+* var arr = emptyLike( x, 'uint8' );
+* // returns <Uint8Array>
+*/
+declare function emptyLike( x: Uint8Array ): Uint8Array;
+
+/**
+* Creates an uninitialized array having the same length as a provided input array.
+*
+* ## Notes
+*
+* -   In browser environments, the function always returns zero-filled arrays.
+* -   In Node.js versions `>=3.0.0`, the underlying memory of returned typed arrays is **not** initialized. Memory contents are unknown and may contain **sensitive** data.
+*
+* @param x - input array from which to derive the output array length
+* @param dtype - data type
+* @returns empty array
+*
+* @example
+* var zeros = require( `@stdlib/array/zeros` );
+*
+* var x = zeros( 2, 'float32' );
+* // returns <Float32Array>[ 0.0, 0.0 ]
+*
+* var arr = emptyLike( x, 'uint8c' );
+* // returns <Uint8ClampedArray>
+*/
+declare function emptyLike( x: Uint8ClampedArray ): Uint8ClampedArray;
+
+/**
+* Creates a zero-filled array having a specified length.
+*
+* @param x - input array from which to derive the output array length
+* @param dtype - data type
+* @returns zero-filled array
+*
+* @example
+* var zeros = require( `@stdlib/array/zeros` );
+*
+* var x = zeros( 2, 'float32' );
+* // returns <Float32Array>[ 0.0, 0.0 ]
+*
+* var arr = emptyLike( x, 'generic' );
+* // returns
+*/
+declare function emptyLike( x: Array<any> ): Array<any>;
```

Comment:
```suggestion
declare function emptyLike( x: Array<any> ): Array<number>;
```

We return a zero-filled generic array.